src/services/EcoBridge/OmniBridge/OmniBridge.ts

```diff
@@ -0,0 +1,763 @@
+        this._account,
+        this._staticProviders[fromToken.chainId]
+      )
+
+      if (!allowance || !fromToken.mode) {
+        this.store.dispatch(
+          ecoBridgeUIActions.setStatusButton({
+            label: 'Cannot fetch allowance. Try later',
+            isError: false,
+            isLoading: false,
+            isBalanceSufficient: false,
+            isApproved: false,
+          })
+        )
+        return
+      }
+
+      if (['NATIVE', 'erc677'].includes(fromToken.mode) || allowance.gte(fromToken.amount)) {
+        this.store.dispatch(
+          ecoBridgeUIActions.setStatusButton({
+            label: 'Bridge',
+            isError: false,
+            isLoading: false,
+            isBalanceSufficient: true,
+            isApproved: true,
+          })
+        )
+      } else {
+        this.store.dispatch(
+          ecoBridgeUIActions.setStatusButton({
+            label: 'Approve',
+            isError: false,
+            isLoading: false,
+            isBalanceSufficient: true,
+            isApproved: false,
+          })
+        )
+      }
+    } catch (e) {
+      this.store.dispatch(
+        ecoBridgeUIActions.setStatusButton({
+          label: 'Cannot execute validate',
+          isError: false,
+          isLoading: false,
+          isBalanceSufficient: true,
+          isApproved: false,
+        })
+      )
+    }
+  }
+
+  public fetchDynamicLists = async () => {
+    try {
+      if (!this._activeChainId) return
+
+      const { from, to } = this.store.getState().ecoBridge.ui
+
+      const selectedChains = [from.chainId, to.chainId]
+
+      if (!(selectedChains.includes(this._foreignChainId) && selectedChains.includes(this._homeChainId))) {
+        this.store.dispatch(this.actions.setTokenListsStatus(SyncState.READY))
+        return
+      }
+
+      this.store.dispatch(this.actions.setTokenListsStatus(SyncState.LOADING))
+
+      const homeEndpoint = getGraphEndpoint(from.chainId, this.bridgeId)
+      const foreignEndpoint = getGraphEndpoint(to.chainId, this.bridgeId)
+
+      const fetchDefaultTokens = async () => {
+        const url = defaultTokensUrl[Number(from.chainId)]
+
+        const tokenListValidator = new Ajv({ allErrors: true }).compile(schema)
+
+        const response = await fetch(url)
+        if (response.ok) {
+          const json: TokenList = await response.json()
+          if (tokenListValidator(json)) {
+            return json.tokens.filter(token => token.chainId === from.chainId)
+          }
+        }
+        return []
+      }
+
+      const [homeData, foreignData, defaultTokens] = await Promise.all<SubgraphResponse, SubgraphResponse, TokenInfo[]>(
+        [request(homeEndpoint, homeTokensQuery), request(foreignEndpoint, foreignTokensQuery), fetchDefaultTokens()]
+      )
+
+      const homeTokens = homeData && homeData.tokens ? homeData.tokens : []
+      const foreignTokens = foreignData && foreignData.tokens ? foreignData.tokens : []
```

Comment:
Promise.all will throw error when one of the promises fails so code won't get executed till this point anyway
```suggestion
      const homeTokens = homeData.tokens
      const foreignTokens = foreignData.tokens
```

Reply:
fixed [a2c2761](https://github.com/levelkdev/swapr-dapp/pull/1007/commits/a2c2761ff4b65a4f7e3d129ecd0d7e484177d0b0)